src/task_dynamics/OneStepToZero.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017-2021 CNRS-AIST JRL and CNRS-UM LIRMM
+ */
+
+#include <tvm/task_dynamics/OneStepToZero.h>
+
+#include <tvm/function/abstract/Function.h>
+
+namespace tvm
+{
+
+namespace task_dynamics
+{
+
+void OneStepToZero::checkParam(Order d, double dt)
+{
+  if(d == Order::Zero)
+    throw std::runtime_error("[task_dynamics::OneStepToZero] This task dynamics only work for order from 1. For order "
+                             "0, maybe task_dynamics::None would fit you need.");
+
+  if(d > Order::Two)
+    throw std::runtime_error("[task_dynamics::OneStepToZero] Implementation only accepts order up to 2.");
+
+  if(dt <= 0)
+    throw std::runtime_error("[task_dynamics::OneStepToZero] dt needs to be positive");
```

Comment:
Please include the faulty parameter in the exception message to make it eventually easier to track down

i.e. include `d`/`dt` in the exception message